internal/freelist/serde.go

```diff
@@ -0,0 +1,79 @@
+package freelist
+
+import (
+	"fmt"
+	"sort"
+	"unsafe"
+
+	"go.etcd.io/bbolt/internal/common"
+)
+
+type Serializable interface {
+	// Read calls Init with the page ids stored in te given page.
+	Read(f Freelist, page *common.Page)
+
+	// Write writes the freelist into the given page.
+	Write(f Freelist, page *common.Page)
+
+	// EstimatedWritePageSize returns the size of the page after serialization in Write.
+	// This should never underestimate the size.
+	EstimatedWritePageSize(f Freelist) int
```

Comment:
Let's try not add something new in this PR to make the review easier. Anything new should be discussed separately
```suggestion
```